src/Handlers/RedisHandler.php

```diff
@@ -0,0 +1,177 @@
+<?php
+
+namespace CodeIgniter\Settings\Handlers;
+
+use CodeIgniter\I18n\Time;
+use Config\Cache;
+use RuntimeException;
+
+/**
+ * Provides cache persistence for Settings.
+ * Uses ArrayHandler for storage to minimize cache calls.
+ */
+class RedisHandler extends ArrayHandler
+{
+    /**
+     * The Query Builder for the Settings table.
+     */
+    private BaseRedisHandler $handler;
+
+    /**
+     * Array of contexts that have been stored.
+     *
+     * @var ?string[]
+     */
+    private $hydrated = [];
```

Comment:
```suggestion
    private ?array $hydrated = [];
```

Reply:
I think the `null` is not needed?

Reply:
See my other comment; I think this is actually `array<string|null>`. I'm not sure if the shorthand is incorrect or if I just don't know about it 😂